src/dmd/dtemplate.d

```diff
@@ -2244,7 +2244,7 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
 
         paramscope.pop();
         //printf("\tmatch %d\n", match);
-        return cast(MATCH)(match | (matchTiargs << 4));
+        return cast(MATCHpair)(match | (matchTiargs << 4));
```

Comment:
Why not make `MATCHpair` a `struct { MATCH, MATCH }` and do away with this bit fiddling done by the callee/caller?

Reply:
I could, but since the packing/unpacking each only occurs in one place, I don't see a benefit for adding another dozen lines of code.

Reply:
did it anyway while looking for the bug